bitnami/thanos/templates/_helpers.tpl

```diff
@@ -209,9 +209,29 @@ thanos: objstore configuration
 
 {{/* Validate values of Thanos - Ruler Alertmanager(s) */}}
 {{- define "thanos.validateValues.ruler.alertmanagers" -}}
-{{- if and .Values.ruler.enabled (empty .Values.ruler.alertmanagers) -}}
+{{/* Check the emptiness of the values */}}
+{{- if and .Values.ruler.enabled ( and (empty .Values.ruler.alertmanagers) (empty .Values.ruler.alertmanagersConfig)) -}}
 thanos: ruler alertmanagers
-    When enabling Ruler component, you must provide alermanagers URL(s).
+    When enabling Ruler component, you must provide either alermanagers URL(s), an alertmanagers configuration, or an alertmanagers configuration file.
+    See https://github.com/thanos-io/thanos/blob/ef94b7e6468d94e2c47943ebf5fc6db24c48d867/docs/components/rule.md#flags and https://github.com/thanos-io/thanos/blob/ef94b7e6468d94e2c47943ebf5fc6db24c48d867/docs/components/rule.md#Configuration for more information.
+{{- end -}}
+{{/* Check that the values are defined in a mutually exclusive manner */}}
+{{- if and .Values.ruler.enabled .Values.ruler.alertmanagers .Values.ruler.alertmanagersConfig -}}
+thanos: ruler alertmanagers
+    Only one of the following can be used at one time:
+        * .Values.ruler.alertmanagers
+        * .Values.ruler.alertmanagersConfig
+    Otherwise, the configurations will collide and Thanos will error out. Please consolidate your configuration
+    into one of the above options.
+{{- else if and .Values.ruler.enabled .Values.ruler.alertmanagersConfig -}}
```

Comment:
In case `alertmanagersConfig` is `nil` we need to check it in an outer if-statement before checking if `alertmanagersConfig.alertmanagers` or `alertmanagersConfig.file` exist or else Go templates will error on accessing a field in a `nil` struct.